src/collection.ts

```diff
@@ -806,13 +806,12 @@ export class Collection<TSchema extends Document = Document> {
       return indexes;
     }
 
-    const info: Record<string, Array<[string, unknown]>> = {};
-    for (const { name, key } of indexes) {
-      info[name] = Object.entries(key);
-    }
+    const object: Record<string, Array<[string, unknown]>> = Object.fromEntries(
```

Comment:
```suggestion
    const object: Record<string, Array<[name: string, direction: IndexDirection]>> =
```

Just leaving a bit more TS context for the dev who fixes NODE-6029